### ISSUE TITLE
Podman-in-docker installation script

### ISSUE THREAD

**Issue by johanbrandhorst** [microsoft/vscode-dev-containers#1042]
We have scripts for `docker-from-docker` and `docker-in-docker`, but now that Docker Inc [has decided it's time to pay](https://thenewstack.io/docker-says-some-users-must-pay-but-can-it-make-money/) it would be cool to investigate whether we could run [podman](https://thenewstack.io/docker-says-some-users-must-pay-but-can-it-make-money/) as a docker replacement within our Dev containers. The answer is of course yes, but it would be great to have a script like `docker-in-podman` for setting up a nice working environment using `podman`.

Relates to: Both

**Comment by Chuxel** [comment]
FYI - Licensing for Docker CE on Linux has not changed. So the "in-docker" part is not affected by the Docker Desktop announcement.

**Comment by johanbrandhorst** [comment]
Sorry, I meant to say `podman-in-docker` consistently. I have edited the original description. Regardless of the licensing situation, it would be nice to have a podman environment script if you are using podman elsewhere because of the licensing changes. Is this not the right repository for this sort of request?

**Comment by Chuxel** [comment]
Yeah for using podman inside a dev container, this is the spot. Just making sure you didn't mean the inverse. 😄

**Comment by PavelSosin-320** [comment]
@johanbrandhorst [How to ... ](https://www.redhat.com/sysadmin/podman-inside-container). I don't think that somebody knows it better. You can ask them.
Running Podman is OK. If you have Podman installed you can use buildah:
buildah from dev container && buildah run apt install  and finally buildah commit.
Running container runtimes runc, crun, cri-o kata, ... inside each other ?????

**Comment by juzuluag** [comment]
Hi, 
The team that I'm working with is interested in this enhancement and I have couple of questions:
- Would be this possible by leveraging `docker-in-docker` case?
- This won't aim for other platforms (windows, macOS) right? where there is no container support.
Any comments/help really appreciate.

**Comment by Chuxel** [comment]
@juzuluag Might be worth explaining a bit what you are looking to do here. The proposed script here, if contributed, would install podman in an Linux container image for use. You'd still need a host the dev container in something on macOS or Linux.
